sympy/solvers/tests/test_solveset.py

```diff
@@ -757,6 +757,13 @@ def test_solve_trig():
         imageset(Lambda(n, 2*n*pi), S.Integers)
 
 
+def test_issue_13084():
+    n = Dummy('n')
+    f1 = sin(2*x)**2 + 2* sin(2*x) + 1
+    f2 = sin(2*x) + 1
+    assert solveset(f1, x, S.Reals) == ImageSet(Lambda(n, n*pi + 3*pi/4), S.Integers)
+    assert solveset(f1, x, S.Reals) == ImageSet(Lambda(n, n*pi + 3*pi/4), S.Integers)
+
```

Comment:
- `solveset(abs(x) - n, x, S.Reals)`
`solveset(sqrt(x) - 1,x,S.Reals)`
 with `Exception RuntimeError: 'maximum recursion depth exceeded' in <generator object <genexpr> at 0x7ff8024375a0> ignored`
If I run the test this exception is raised, otherwise(in seperate file) it is passing.(I'm not able to understand why this is happening.)

- `decompogen(2**x - 10)` raises `sympy.polys.polyerrors.GeneratorsNeeded: can't initialize from 'dict' without generators`. 
I guess this is `decompogen's` issue. Should I open an issue?